plugins/woocommerce-blocks/assets/js/blocks/add-to-cart-with-options/grouped-product-selector/product-item-template/edit.tsx

```diff
@@ -0,0 +1,167 @@
+/**
+ * External dependencies
+ */
+import { useState } from '@wordpress/element';
+import { store as coreStore } from '@wordpress/core-data';
+import {
+	InnerBlocks,
+	useBlockProps,
+	store as blockEditorStore,
+	__experimentalUseBlockPreview as useBlockPreview,
+	BlockContextProvider,
+} from '@wordpress/block-editor';
+import { BlockInstance, type BlockEditProps } from '@wordpress/blocks';
+import { withProduct } from '@woocommerce/block-hocs';
+import {
+	ProductDataContextProvider,
+	useProductDataContext,
+} from '@woocommerce/shared-context';
+import { useSelect } from '@wordpress/data';
+import type { ProductResponseItem } from '@woocommerce/types';
+
+interface Attributes {
+	className?: string;
+}
+
+type ProductItemProps = {
+	attributes: { productId: number; productType: string };
+	isLoading?: boolean;
+	product?: ProductResponseItem;
+	blocks: BlockInstance[];
+	isSelected: boolean;
+	onSelect(): void;
+};
+
+const ProductItem = withProduct( function ProductItem( {
+	attributes,
+	isLoading,
+	product,
+	blocks,
+	isSelected,
+	onSelect,
+}: ProductItemProps ) {
+	const blockPreviewProps = useBlockPreview( {
+		blocks,
+	} );
+
+	return (
+		<BlockContextProvider
+			value={ {
+				postId: attributes.productId,
+				postType: attributes.productType,
+			} }
+		>
+			<ProductDataContextProvider
+				product={ product as ProductResponseItem }
+				isLoading={ isLoading as boolean }
+			>
+				{ isSelected ? (
+					<InnerBlocks
+						templateLock="all"
```

Comment:
No need to do it in this PR, but I think we could allow reordering the blocks (ie, moving the Quantity Selector to the end of the row).